### ISSUE TITLE
segfault in test_libdynd

### ISSUE THREAD

**Issue by jreback** [libdynd/libdynd#258]
just built from master on macosx

v0.6.6-464-g501d054

``
[----------] 5 tests from TypeCasting
[ RUN      ] TypeCasting.IsLosslessAssignment
[       OK ] TypeCasting.IsLosslessAssignment (0 ms)
[ RUN      ] TypeCasting.StringToInt32
[       OK ] TypeCasting.StringToInt32 (0 ms)
[ RUN      ] TypeCasting.StringToInt64
[       OK ] TypeCasting.StringToInt64 (0 ms)
[ RUN      ] TypeCasting.StringToInt128
[       OK ] TypeCasting.StringToInt128 (0 ms)
[ RUN      ] TypeCasting.StringToUInt64
[       OK ] TypeCasting.StringToUInt64 (0 ms)
[----------] 5 tests from TypeCasting (0 ms total)

[----------] 5 tests from SubstituteTypeVars
[ RUN      ] SubstituteTypeVars.SimpleNoSubstitutions
Segmentation fault: 11
```

lmk what other info I can provide

**Comment by mwiebe** [comment]
Yeah, good idea about printing the version/platform info at the beginning. We don't have an OSX CI setup, so it doesn't surprise me that would break. I think also the behaviour across OSX versions varies considerably. Do you know how to get a stack trace from the segfault? I think you can do it with lldb.

**Comment by izaid** [comment]
My main platform is OS X 10.10, things are working here.

**Comment by izaid** [comment]
Interestingly, I just tried to upgrade to CUDA 7.0, which requires clang instead of GCC. I now get this test failure.

To be clear, I was building with clang before as well when I didn't use CUDA (up to earlier today), but it wasn't an issue.

**Comment by izaid** [comment]
FYI, it is just two tests. The rest of DyND works.

**Comment by jreback** [comment]
FYI I am running 10.9

**Comment by izaid** [comment]
While I haven't fixed this, I have modified test_libdynd to simply fail (for those two tests) instead of segault on Mac. You should be able to run the rest of test_libdynd without any problems.

@mwiebe It's strange that there is only an issue in two type substitution tests. If it was really a problem with type substitution, we should get failures everywhere.

**Comment by jreback** [comment]
Cool...that finishes now thanks!

i'll leave the issue open so the failed tests are noted

```
[----------] Global test environment tear-down
[==========] 526 tests from 95 test cases ran. (221 ms total)
[  PASSED  ] 524 tests.
[  FAILED  ] 2 tests, listed below:
[  FAILED  ] SubstituteTypeVars.SimpleNoSubstitutions
[  FAILED  ] SubstituteTypeVars.SimpleSubstitution

 2 FAILED TESTS
```
